pcsx2/Recording/InputRecordingControls.cpp

```diff
@@ -201,4 +206,21 @@ void InputRecordingControls::Lock(u32 frame)
 	if (!g_InputRecording.GetInputRecordingData().FromSaveState())
 		g_FrameCount = frame + 1;
 }
+
+bool InputRecordingControls::IsReplayingAtEnd() const
+{
+	return g_InputRecording.IsReplaying() && g_InputRecording.GetFrameCounter() >= g_InputRecording.GetInputRecordingData().GetTotalFrames();
+}
+
+void InputRecordingControls::EndVideoCapture() const
+{
+	if (MainEmuFrame* mainFrame = GetMainFramePtr())
+	{
+		if (mainFrame->IsCapturing())
+		{
+			mainFrame->VideoCaptureToggle();
+			recordingConLog(L"Video Capture Completed\n");
+		}
+	}
```

Comment:
could be combined into a single `if`

Reply:
This style of nesting the second condition after doing something similar to `MainEmuFrame* mainFrame = GetMainFramePtr()` matches other iterations in the solution. Seems a bit easier to follow readability-wise as well.

Reply:
oh my mistake, i didn't notice it was doing assignment in the if statement.  This could be minimized down using the `sMainFrame` macro I believe.  Something like `if (sMainFrame.IsCapturing())`.  But I don't have a strong opinion this _should_ also safeguard against an uninitialized mainframe.

Reply:
That was my first attempt. Didn't work so I had to do this.